proxy.go

```diff
@@ -56,6 +56,17 @@ func isCloseable(err error) bool {
 
 // Proxy is an HTTP proxy with support for TLS MITM and customizable behavior.
 type Proxy struct {
+	roundTripper http.RoundTripper
+	dial         func(string, string) (net.Conn, error)
+	mitm         *mitm.Config
+	proxyURL     func(*http.Request) (*url.URL, error)
+	conns        sync.WaitGroup
+	connsMu      sync.Mutex // protects conns.Add/Wait from concurrent access
+	closing      chan bool
+
+	reqmod RequestModifier
+	resmod ResponseModifier
+
```

Comment:
Why do we want to have not exported fields first? 
For example http server https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/net/http/server.go;l=2590 have exported fields first.

Reply:
I dropped this commit.